application/generator_test.go

```diff
@@ -0,0 +1,84 @@
+package application
+
+import (
+	"config-manager/domain"
+	"config-manager/utils"
+	"strings"
+	"testing"
+)
+
+func TestGenerateAllEnabled(t *testing.T) {
+	templates := utils.FilesIntoMap("../playbooks/test/", "*.yml")
+
+	pbg := &Generator{
+		Templates: templates,
+	}
+
+	state := domain.StateMap{
+		"test1": "enabled",
+		"test2": "enabled",
+	}
+
+	pb, err := pbg.GeneratePlaybook(state)
+	if err != nil {
+		t.Error(err)
+	}
+
+	expectedPlay1 := `
+- name: Test 1
+  hosts: localhost
+  tasks:
+    - name: Print 1 setup
+      debug:
+        msg: "1 setup"`
+
+	expectedPlay2 := `
+- name: Test 2
+  hosts: localhost
+  tasks:
+    - name: Print 2 setup
+      debug:
+        msg: "2 setup"`
+
+	if !strings.Contains(pb, expectedPlay1) || !strings.Contains(pb, expectedPlay2) {
```

Comment:
This is a reasonable approach, but you might think about using "golden files" in the future.  It might make the tests a bit easier to read/write.  Just a suggestion for future consideration.

Reply:
Agreed. I'm planning to rework the structure of these tests in a follow up PR